operator/v1/types.go

```diff
@@ -284,6 +288,10 @@ type KlusterletSpec struct {
 	// DeployOption contains the options of deploying a klusterlet
 	// +optional
 	DeployOption KlusterletDeployOption `json:"deployOption,omitempty"`
+
+	// SpokeRegistrationFeatureGates represents the list of feature gates for spoke registration
+	// +optional
+	SpokeRegistrationFeatureGates []string `json:"spokeRegistrationFeatureGates,omitempty"`
```

Comment:
> this will need more consideration and document.
> 
> 1. if this is not set, default feature gates will be used.
> 2. If this is set, will it replace all default feature gates or it overrides the default feature gates?
> 3. What should user inputs? is it as `["featuregate1=true", "featuregate2=false"]`?

I will add more consideration and document.